### PR TITLE
Fix: Added middleware and passed request headers down from buffer for websockets

### PR DESCRIPTION
**Pull request by jsun-m**
The TaskMiddleware used the BasicMiddleware class which happened to skip the dispatch function if the type of http connection is a websocket conenction. This adds a new middleware to handle the task lifecycle functionality for just websockets

### CODE REVIEW

**Review by luke-lombardi — CHANGES_REQUESTED** [review]
Can we move the middleware out to a separate file? Feels like it's getting really messy

File: sdk/src/beta9/runner/endpoint.py [hunk 1]
```diff
@@ -127,51 +128,67 @@ class TaskLifecycleData:
     override_callback_url: Optional[str] = None
 
 
-class TaskLifecycleMiddleware(BaseHTTPMiddleware):
-    async def dispatch(self, request: Request, call_next):
-        if request.url.path == "/health":
-            return await call_next(request)
+async def runTask(request, func, func_args):
```
Comment (luke-lombardi) [inline sdk/src/beta9/runner/endpoint.py:131]: let's stick to style conventions here `handle_task` maybe

File: sdk/src/beta9/runner/endpoint.py [hunk 2]
```diff
@@ -209,6 +226,7 @@ def __init__(self, logger: logging.Logger, worker: UvicornWorker) -> None:
             self.app = FastAPI(lifespan=self.lifespan)
 
         self.app.add_middleware(TaskLifecycleMiddleware)
+        self.app.add_middleware(WebsocketTaskLifecycleMiddleware)
```
Comment (luke-lombardi) [inline sdk/src/beta9/runner/endpoint.py:229]: I guess we need to add both - they don't conflict do they? Like we're not going to accidentally be using the websocket middleware for non-websockets. Also another concern is overhead of two layers of middleware, are they both executed (or attempted to be executed) everytime?

Comment (jsun-m) [reply sdk/src/beta9/runner/endpoint.py:158]: both are being executed but it immediately skips the websocket one cause of this
```
if scope["type"] != "websocket":
            await self.app(scope, receive, send)
            return

```

The TaskMiddleware behind the scenes already do
```
if scope["type"] != "http":
            await self.app(scope, receive, send)
            return
```

**Review by luke-lombardi — APPROVED** [review]

**Pull request merged**